pallets/settlement/src/lib.rs

```diff
@@ -904,10 +908,27 @@ impl<T: Trait> Module<T> {
         Ok(instruction_counter)
     }
 
+    fn get_valid_legs(
+        instruction_id: u64,
+        legs_count: u64,
+        portfolios: &BTreeSet<PortfolioId>,
+    ) -> Result<Vec<(u64, Leg<T::Balance>)>, DispatchError> {
+        let legs = <InstructionLegs<T>>::iter_prefix(instruction_id)
+            .filter(|(_, leg_details)| portfolios.contains(&leg_details.from))
+            .collect::<Vec<(u64, Leg<T::Balance>)>>();
+        // Ensure given legs count is matches with the total number of legs given portfolio have.
+        ensure!(
+            legs.len() as u64 == legs_count,
+            Error::<T>::IncorrectLegCountSubmitted
+        );
```

Comment:
We should allow overpaying. I, for one, will mostly be overpaying while testing xD.

```suggestion
        ensure!(
            legs.len() as u64 <= legs_count,
            Error::<T>::IncorrectLegCountSubmitted
        );
```

Reply:
Maybe rename the error variant then also? `LegCountTooSmall`?

Reply:
It could potentially be beneficial to quit early here, doing something like:
```rust
        let mut legs_iter = <InstructionLegs<T>>::iter_prefix(instruction_id)
            .filter(|(_, leg_details)| portfolios.contains(&leg_details.from));
        let legs = legs_iter
            .by_ref()
            .take(legs_count)
            .collect::<Vec<(u64, Leg<T::Balance>)>>();
        ensure!(legs_iter.next().is_none(), Error::<T>::IncorrectLegCountSubmitted);
```
This will probably make successful calls slightly more expensive though.

Reply:
@Centril Let's keep as it is, It is simple and less expensive and also covers the overpaying 😉

Reply:
Yeah I'm down with that since its not a clear win 😃, but what about the renaming bit?

Reply:
pushed